server.go

```diff
@@ -95,39 +105,47 @@ func (s *SockchatServer) webSocket(w http.ResponseWriter, r *http.Request) {
 			if os.IsTimeout(err) {
 				conn.WriteSocketMsg(NewSocketMessage("connection_timed_out", "{}"))
 			}
-			log.Printf("error occured when listening for ws messages, closing connection: %v", err)
-			s.shutConnection(conn)
+			if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure) {
+				log.Printf("error: %v", err)
+			}
 			break
 		}
-		if receivedMsg.Action == LoginAction {
-			ctx, cancel := context.WithTimeout(r.Context(), ResponseDeadline)
-			defer cancel()
-			s.loginUser(ctx, *receivedMsg, conn)
-		} else {
-			if conn.nick == "" {
-				conn.WriteSocketMsg(NewErrorMessage("Log in first using " + LoginAction))
-			}
-			conn.SetReadDeadline(time.Now().Add(s.timeoutAuthorized))
-			switch receivedMsg.Action {
-			case CreateAction:
-				s.createNewChannel(*receivedMsg, conn)
-			case JoinAction:
-				s.joinChannel(*receivedMsg, conn)
-			case LeaveAction:
-				s.leaveChannel(*receivedMsg, conn)
-			case SendMessageAction:
-				s.sendMessageToChannel(*receivedMsg, conn)
-			default:
-				log.Printf("Unexpected request received: %s", receivedMsg.Action)
-				conn.WriteSocketMsg(NewErrorMessage("action not supported"))
+		ctx, cancel := context.WithTimeout(r.Context(), ResponseDeadline)
+		defer cancel()
```

Comment:
Move this context to `authorizeConnection` method, currently contexts are created with every message and they are not closed until the whole `webSocket` loop finishes